### PR TITLE
Improve the description of how to provide addons documentation

### PR DESCRIPTION
**Pull request by klaudiagrz**
**Description**

Changes proposed in this pull request:

- Improve the description of how to provide addons documentation
- Update the description of addons structure

### CODE REVIEW

File: docs/helm-broker/03-01-create-addons.md [hunk 1]
```diff
@@ -95,29 +95,27 @@ The `plans` directory must contain at least one plan. Each plan must contain the
...
+| **docs[].template.description**        |   YES   | Provides the description of the ClusterDocsTopic or DocsTopic. |
 | **docs[].template.sources[]**          |   YES   | Contains the definitions of assets for an addon. |
 | **docs[].template.sources[].type**     |   YES   | Defines the type of the asset. |
-| **docs[].template.sources[].name**     |   YES   | Defines the name of the asset. |
-| **docs[].template.sources[].mode**     |   YES   | Defines the type of the asset file. |
-| **docs[].template.sources[].url**      |   YES   | Defines the URL under which the asset is stored. |
-| **docs[].template.sources[].filter**   |   YES   | Defines the path from which to upload assets.  |
+| **docs[].template.sources[].name**     |   YES   | Defines a unique identifier of a given asset. It must be unique if there is more than one asset of a given type. |
+| **docs[].template.sources[].mode**     |   YES   | Specifies if the asset consists of one file or a set of compressed files in the ZIP or TAR format. Use `single` for one file and `package` for a set of files. |
+| **docs[].template.sources[].url**      |   YES   | Specifies the location of a single file or a package. |
```
Comment (polskikiel) [inline docs/helm-broker/03-01-create-addons.md]:
```suggestion
| **docs[].template.sources[].url**      |   YES   | Specifies the location of a file. |
```

File: docs/helm-broker/03-01-create-addons.md [hunk 2]
```diff
@@ -95,29 +95,27 @@ The `plans` directory must contain at least one plan. Each plan must contain the
...
+| **docs[].template**                    |   YES   | Contains the specification of the ClusterDocsTopic or DocsTopic. |
+| **docs[].template.displayName**        |   YES   | Specifies the display name of the ClusterDocsTopic or DocsTopic. |
+| **docs[].template.description**        |   YES   | Provides the description of the ClusterDocsTopic or DocsTopic. |
 | **docs[].template.sources[]**          |   YES   | Contains the definitions of assets for an addon. |
 | **docs[].template.sources[].type**     |   YES   | Defines the type of the asset. |
-| **docs[].template.sources[].name**     |   YES   | Defines the name of the asset. |
-| **docs[].template.sources[].mode**     |   YES   | Defines the type of the asset file. |
-| **docs[].template.sources[].url**      |   YES   | Defines the URL under which the asset is stored. |
-| **docs[].template.sources[].filter**   |   YES   | Defines the path from which to upload assets.  |
+| **docs[].template.sources[].name**     |   YES   | Defines a unique identifier of a given asset. It must be unique if there is more than one asset of a given type. |
```
Comment (polskikiel) [inline docs/helm-broker/03-01-create-addons.md]:
```suggestion
| **docs[].template.sources[].name**     |   YES   | Defines a unique identifier of a given asset. It must be unique in a given type. |
```

File: docs/helm-broker/03-01-create-addons.md [hunk 3]
```diff
@@ -95,29 +95,27 @@ The `plans` directory must contain at least one plan. Each plan must contain the
...
+| **docs[].template.sources[].mode**     |   YES   | Specifies if the asset consists of one file or a set of compressed files in the ZIP or TAR format. Use `single` for one file and `package` for a set of files. |
+| **docs[].template.sources[].url**      |   YES   | Specifies the location of a single file or a package. |
+| **docs[].template.sources[].filter**   |   YES   | Specifies a set of assets from the package to upload. The regex used in the filter must be [RE2](https://golang.org/s/re2syntax)-compliant.  |
```
Comment (polskikiel) [inline docs/helm-broker/03-01-create-addons.md]: Specifies the folder from which the documentation is fetched

File: docs/helm-broker/03-07-addons-docs.md [hunk 4]
```diff
@@ -1,41 +1,56 @@
...
+  </summary>
+
+In the `meta.yaml` file, provide the **url** parameter with a value that points to the address where the documentation is stored.
+
+  </details>
+</div>
+
+## Provide documentation for objects
+
+To provide documentation for objects that appear after provisioning your addon, create the `docs.yaml` file inside the addon's chart. This file contains [ClusterDocsTopic](/components/headless-cms/#custom-resource-cluster-docs-topic) or [DocsTopic](/components/headless-cms/#custom-resource-docstopic) custom resources. Each ClusterDocsTopic or DocsTopic corresponds to a single object with the same ID as the name of the specified object. Your `docs.yaml` file can contain many ClusterDocsTopics or DocsTopics.
```
Comment (polskikiel) [inline docs/helm-broker/03-07-addons-docs.md]: a yaml file, the name `docs.yaml` is not mandatory

File: docs/helm-broker/03-01-create-addons.md [hunk 5]
```diff
@@ -95,29 +95,27 @@ The `plans` directory must contain at least one plan. Each plan must contain the
...
-As the Helm Broker is installed as a ClusterServiceBroker, documentation for addons is provided using [ClusterDocsTopics](/components/headless-cms/#custom-resource-clusterdocstopic).
+In the `docs` directory, provide documentation for your addon. The documentation can include Markdown documents, AsyncAPI, OData, and OpenAPI specification files. Create the `assets` directory inside the `docs` directory to store assets, such as images. The `docs` directory must contain a `meta.yaml` file which provides information on how documentation for the addon is uploaded.
+As you can install the Helm Broker as a ClusterServiceBroker or as a ServiceBroker, documentation for addons is provided using either [ClusterDocsTopics](/components/headless-cms/#custom-resource-clusterdocstopic) or [DocsTopics](/components/headless-cms/#custom-resource-docs-topic) custom resources, respectively.
 
 The `meta.yaml` file contains the specification of the ClusterDocsTopic. The example structure of the `meta.yaml` file looks as follows:
```
Comment (polskikiel) [inline docs/helm-broker/03-01-create-addons.md]: ClusterDocsTopic or DocsTopics -:> be consistent

OR use only DocsTopics to minimalize the amount of the text

File: docs/helm-broker/03-07-addons-docs.md (hunk 4 above)
Comment (klaudiagrz) [reply docs/helm-broker/03-07-addons-docs.md]: I guess it's the same with `meta.yaml` and `docs` directory?

File: docs/helm-broker/03-01-create-addons.md (hunk 5 above)
Comment (klaudiagrz) [reply docs/helm-broker/03-01-create-addons.md]: Good point! 🤔

File: docs/helm-broker/03-03-create-addons-repo.md [hunk 6]
```diff
@@ -89,6 +89,8 @@ sample-addon-repository
 
 See the example of the Kyma `addons` repository [here](https://github.com/kyma-project/addons/tree/master/addons).
 
+> **NOTE:** The maximum size of your addons repository is determined by the amount of memory and storage size. These limits are set in the
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply docs/helm-broker/03-03-create-addons-repo.md]: GJ!

File: docs/helm-broker/03-07-addons-docs.md [hunk 7]
```diff
@@ -1,41 +1,56 @@
...
-
-One ClusterDocsTopic or DocsTopic object corresponds to a single Service Class with the same ID as the name of the specified object. Store documentation for each Service Class in the `docs/{class_name}` directory which corresponds to the value of the **filter** parameter in the ClusterDocsTopic or DocsTopic definition.
-
-During the provisioning process, the Helm Broker pushes the **addonRepositoryURL** variable into the chart. The **addonsRepositoryURL** points to your addon compressed to a `.tgz` file.
-
-### Documentation structure
-
-Deliver documentation for your addons in Markdown files with specified metadata. To learn more about the metadata and content of the Markdown files, read [this](/components/headless-cms/#details-markdown-documents) document.
+Using the Helm Broker, you can provide documentation for your addon and display it in the Console UI. There are two cases in which you may want to use this feature:
+- Provide documentation for your addon
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply docs/helm-broker/03-07-addons-docs.md]: I'd stay with `two cases in which` and change `provide` to `providing`, thanks :)

File: docs/helm-broker/03-07-addons-docs.md [hunk 8]
```diff
@@ -1,41 +1,56 @@
...
+- Provide documentation for objects that appear after provisioning your addon
+
+>**NOTE:** Deliver documentation for your addons in Markdown files with specified metadata. To learn more about the metadata and content of the Markdown files, read [this](/components/headless-cms/#details-markdown-documents) document. For more information about currently supported types of the assets, read [this](/components/headless-cms/#overview-overview-headless-cms-in-kyma) document.
+
+## Provide documentation for your addon
+
+To provide documentation for your addon, create the `docs` folder inside it. Your `docs` folder must contain a [`meta.yaml`](#details-create-addons-docs-directory) file with metadata information about how documentation for the addon is uploaded. You can either provide your own documents or point to the external URL with the source documentation:
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply docs/helm-broker/03-07-addons-docs.md]: Yes, exactly! Thanks for spotting that, that's obviously a mental shortcut 😉

File: docs/helm-broker/03-01-create-addons.md [hunk 9]
```diff
@@ -95,29 +92,27 @@ The `plans` directory must contain at least one plan. Each plan must contain the
...
+| **docs[].template**                    |   Yes   | Contains the specification of the ClusterDocsTopic or DocsTopic. |
+| **docs[].template.displayName**        |   Yes   | Specifies the display name of the ClusterDocsTopic or DocsTopic. |
+| **docs[].template.description**        |   Yes   | Provides the description of the ClusterDocsTopic or DocsTopic. |
 | **docs[].template.sources[]**          |   Yes   | Contains the definitions of assets for an addon. |
 | **docs[].template.sources[].type**     |   Yes   | Defines the type of the asset. |
-| **docs[].template.sources[].name**     |   Yes   | Defines the name of the asset. |
-| **docs[].template.sources[].mode**     |   Yes   | Defines the type of the asset file. |
-| **docs[].template.sources[].url**      |   Yes   | Defines the URL under which the asset is stored. |
-| **docs[].template.sources[].filter**   |   Yes   | Defines the path from which to upload assets.  |
+| **docs[].template.sources[].name**     |   Yes   | Defines a unique identifier of a given asset. It must be unique in a given type. |
```
Comment (majakurcius) [inline docs/helm-broker/03-01-create-addons.md]:
```suggestion
| **docs[].template.sources[].name**     |   Yes   | Defines a unique identifier of a given asset. It must be unique for a given type. |
```

Is that what you mean? I'm not sure what the `type` refers to.

**Pull request merged**